prow/config/tide.go

```diff
@@ -221,34 +291,57 @@ func (qm QueryMap) ForRepo(org, repo string) TideQueries {
 // * a label that is in both the labels and missing_labels section
 // * a branch that is in both included and excluded branch set.
 func (tq *TideQuery) Validate() error {
+	orgs, repos, excludedRepos := sets.NewString(), sets.NewString(), sets.NewString()
 	for o := range tq.Orgs {
 		if strings.Contains(tq.Orgs[o], "/") {
 			return fmt.Errorf("orgs[%d]: %q contains a '/' which is not valid", o, tq.Orgs[o])
 		}
 		if len(tq.Orgs[o]) == 0 {
 			return fmt.Errorf("orgs[%d]: is an empty string", o)
 		}
+		orgs.Insert(tq.Orgs[o])
+	}
+	if dups := len(tq.Orgs) - orgs.Len(); dups != 0 {
+		return fmt.Errorf("'orgs' contains %d duplicate entries", dups)
 	}
 
 	for r := range tq.Repos {
 		parts := strings.Split(tq.Repos[r], "/")
 		if len(parts) != 2 || len(parts[0]) == 0 || len(parts[1]) == 0 {
 			return fmt.Errorf("repos[%d]: %q is not of the form \"org/repo\"", r, tq.Repos[r])
 		}
-		for o := range tq.Orgs {
-			if tq.Orgs[o] == parts[0] {
-				return fmt.Errorf("repos[%d]: %q is already included via orgs[%d]: %q", r, tq.Repos[r], o, tq.Orgs[o])
-			}
+		if orgs.Has(parts[0]) {
+			return fmt.Errorf("repos[%d]: %q is already included via org: %q", r, tq.Repos[r], parts[0])
+		}
+		repos.Insert(tq.Repos[r])
+	}
+	if dups := len(tq.Repos) - repos.Len(); dups != 0 {
+		return fmt.Errorf("'repos' contains %d duplicate entries", dups)
+	}
+
+	for er := range tq.ExcludedRepos {
+		parts := strings.Split(tq.ExcludedRepos[er], "/")
+		if len(parts) != 2 || len(parts[0]) == 0 || len(parts[1]) == 0 {
+			return fmt.Errorf("excludedRepos[%d]: %q is not of the form \"org/repo\"", er, tq.ExcludedRepos[er])
+		}
+		if !orgs.Has(parts[0]) {
+			return fmt.Errorf("excludedRepos[%d]: %q has no effect because org %q is not included", er, tq.ExcludedRepos[er], parts[0])
```

Comment:
It looks like we're adding a lot more cases to Validate, but I don't see that we're adding any more unit tests to ensure these trip as intended

Reply:
Turns out there weren't any unit tests for `TideQuery` validation, it just looked like there were because a test function was misnamed...
I added tests for all the checks that the validation does.